src/coreclr/src/jit/optcse.cpp

```diff
@@ -2703,28 +2818,165 @@ class CSE_Heuristic
                 else if (currVN != firstVN)
                 {
                     allSame = false;
-                    break;
+                }
+
+                BasicBlock*                blk       = lst->tslBlock;
+                const BasicBlock::weight_t curWeight = blk->getBBWeight(m_pCompiler);
+                if (isConstCSE)
```

Comment:
Nit: it would be nice to shrink this function into smaller in the future, it was already 500+ lines, not it is 800+,